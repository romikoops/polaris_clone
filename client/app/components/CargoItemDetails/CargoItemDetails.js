import React, { Component } from 'react';
import styles from './CargoItemDetails.scss';
import PropTypes from 'prop-types';
import { HsCodeViewer } from '../HsCodes/HsCodeViewer';

export class CargoItemDetails extends Component {
    constructor(props) {
        super(props);
        this.state =  {
            viewer: false
        };
        this.viewHsCodes = this.viewHsCodes.bind(this);
    }
    viewHsCodes()  {
        this.setState({
            viewer: !this.state.viewer
        });
        console.log(this.state.viewer);
    }
    render() {
        const { index, item, hsCodes, theme } = this.props;
        const viewer = this.state.viewer;
        const textStyle = {
            background: theme && theme.colors ? '-webkit-linear-gradient(left, ' + theme.colors.primary + ',' + theme.colors.secondary + ')' : 'black'
        };

        return (
            <div className={`${styles.info} layout-row flex-100 layout-wrap layout-align-center`}>
                <div className="flex-100">
                    <h4>Unit {index + 1}</h4>
                </div>
                <hr/>
                <div className="flex-100 layout-row layout-align-space-between">
                    <p>Gross Weight</p>
                    <p>{item.payload_in_kg} kg</p>
                </div>
                <hr className="flex-100"/>
                <div className="flex-100 layout-row layout-align-space-between">
                    <p>Length</p>
                    <p>{item.dimension_y} cm</p>
                </div>
                <hr className="flex-100"/>
                <div className="flex-100 layout-row layout-align-space-between">
                    <p>Width</p>
                    <p>{item.dimension_x} cm</p>
                </div>
                <hr className="flex-100"/>
                <div className="flex-100 layout-row layout-align-space-between">
                    <p>Height</p>
                    <p>{item.dimension_z} cm</p>
                </div>
                <hr className="flex-100"/>
                <div className="flex-100 layout-row layout-align-space-between">
                    <p>Volume</p>
                    <p>{(item.dimension_y * item.dimension_x * item.dimension_y) / 1000000} m<sup>3</sup></p>
                </div>
                <hr className="flex-100"/>
                <div className="flex-100 layout-row layout-wrap" onClick={this.viewHsCodes}>
                    <i className="fa fa-eye clip flex-none" style={textStyle} />
                    <p className="offset-5 flex-none">View Hs Codes</p>
                </div>
                { viewer ? <HsCodeViewer item={item} hsCodes={hsCodes} theme={theme} close={this.viewHsCodes}/> : ''}
            </div>
        );
    }
}
CargoItemDetails.propTypes = {
    item: PropTypes.object,
    index: PropTypes.number
};
