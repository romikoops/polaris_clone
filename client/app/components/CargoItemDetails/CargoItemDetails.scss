:local {
  .info {
    color: #2C2C2C;
    // font-family: ".SF NS Text";
    font-size: 16px;
    line-height: 18px;
		hr {
			border: 0.5px solid rgba(black, .03);
		}
		p {
			margin: 0;
			height: 18px;
			sup {
				font-size: 10px;
			}
		}
  }
}