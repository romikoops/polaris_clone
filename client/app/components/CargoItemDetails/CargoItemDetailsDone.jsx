import React, { Component } from 'react'
import { translate } from 'react-i18next'
import styles from './CargoItemDetails.scss'
import PropTypes from '../../prop-types'
import { HsCodeViewer } from '../HsCodes/HsCodeViewer'
import { trim, WRAP_ROW } from '../../classNames'

const CONTAINER = trim(`
  CARGO_ITEM_DETAILS
  ${styles.info} 
  ${WRAP_ROW(100)} 
  layout-align-center
`)
const LONG_ROW = 'flex-100 layout-row layout-align-space-between'

export class CargoItemDetails extends Component {
  constructor (props) {
    super(props)
    this.state = {
      viewer: false
    }
    this.viewHsCodes = this.viewHsCodes.bind(this)
  }
  viewHsCodes () {
    this.setState({
      viewer: !this.state.viewer
    })
  }
  render () {
    const { viewer } = this.state
    const {
      hsCodes,
      index,
      item,
      t,
      theme,
      viewHSCodes
    } = this.props

    const textStyle = textStyleFn(theme)
    // const HSCodes = () => {
    //   if (!viewHSCodes) {
    //     return ''
    //   }

    //   return (
    //     <div className={WRAP_ROW(100)} onClick={this.viewHsCodes}>
    //       <i className="fa fa-eye clip flex-none" style={textStyle} />
    //       <p className="offset-5 flex-none">View Hs Codes</p>
    //     </div>
    //   )
    // }
    // const Viewer = () => {
    //   if (!viewer) {
    //     return ''
    //   }

    //   return (
    //     <HsCodeViewer
    //       item={item}
    //       hsCodes={hsCodes}
    //       theme={theme}
    //       close={this.viewHsCodes}
    //     />
    //   )
    // }

    const dimensions = dimensionsFn(item)

    return (
      <div className={CONTAINER}>
        <div className="flex-100">
          <h4>{t('common:unit')} {index + 1}</h4>
        </div>

        <hr />

        <div className={LONG_ROW}>
          <p>{t('common:grossWeight')}</p>
          <p>{item.payload_in_kg} kg</p>
        </div>

        <hr className="flex-100" />

        <div className={LONG_ROW}>
          <p>{t('common:length')}</p>
          <p>{item.dimension_y} cm</p>
        </div>

        <hr className="flex-100" />

        <div className={LONG_ROW}>
          <p>{t('common:width')}</p>
          <p>{item.dimension_x} cm</p>
        </div>

        <hr className="flex-100" />

        <div className={LONG_ROW}>
          <p>{t('common:height​​')}</p>
          <p>{item.dimension_z} cm</p>
        </div>

        <hr className="flex-100" />

        <div className={LONG_ROW}>
          <p>{t('common:volume')}</p>
          <p>
            {dimensions} m<sup>3</sup>
          </p>
        </div>

        <hr className="flex-100" />

        <div className={LONG_ROW}>
          <p>{t('common:chargeableWeight')}</p>
          <p>
            {(item.chargeable_weight).toFixed(2)} kg
          </p>
        </div>
        <hr className="flex-100" />
        {viewHSCodes ? (
          <div className="flex-100 layout-row layout-wrap" onClick={this.viewHsCodes}>
            <i className="fa fa-eye clip flex-none" style={textStyle} />
            <p className="offset-5 flex-none">{t('common:viewHsCodes')}</p>
          </div>
        ) : (
          ''
        )}
        {viewer ? (
          <HsCodeViewer item={item} hsCodes={hsCodes} theme={theme} close={this.viewHsCodes} />
        ) : (
          ''
        )}
      </div>
    )
  }
}
CargoItemDetails.propTypes = {
  t: PropTypes.func.isRequired,
  item: PropTypes.shape({
    hs_codes: PropTypes.array
  }).isRequired,
  index: PropTypes.number.isRequired,
  viewHSCodes: PropTypes.bool,
  theme: PropTypes.theme,
  hsCodes: PropTypes.arrayOf(PropTypes.string).isRequired
}

CargoItemDetails.defaultProps = {
  viewHSCodes: false,
  theme: false
}

function textStyleFn (theme) {
  return {
    background:
      theme && theme.colors
        ? `-webkit-linear-gradient(left, ${
          theme.colors.primary
        },${
          theme.colors.secondary
        })`
        : 'black'
  }
}

function dimensionsFn (item) {
  const dimensions = item.dimension_y * item.dimension_x * item.dimension_y

  return (dimensions / 1000000).toFixed(2)
}

export default translate('common')(CargoItemDetails)
