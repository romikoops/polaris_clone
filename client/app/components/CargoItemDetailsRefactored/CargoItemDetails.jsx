import React, { Component } from 'react'
import styles from './CargoItemDetails.scss'
import PropTypes from '../../prop-types'
import { HsCodeViewer } from '../HsCodes/HsCodeViewer'
import { trim, ROW, WRAP_ROW } from '../../classNames'

const CONTAINER = trim(`
  CARGO_ITEM_DETAILS
  ${styles.info} 
  ${WRAP_ROW(100)} 
  layout-align-center
`)
const LONG_ROW = 'flex-100 layout-row layout-align-space-between'

export class CargoItemDetails extends Component {
  constructor (props) {
    super(props)
    this.state = {
      viewer: false
    }
    this.viewHsCodes = this.viewHsCodes.bind(this)
  }
  viewHsCodes () {
    this.setState({
      viewer: !this.state.viewer
    })
  }
  render () {
    const { viewer } = this.state
    const {
      hsCodes,
      index,
      item,
      theme,
      viewHSCodes
    } = this.props

    const textStyle = textStyleFn(theme)
    const HSCodes = () => {
      if (!viewHSCodes) {
        return ''
      }

      return (
        <div className={WRAP_ROW(100)} onClick={this.viewHsCodes}>
          <i className="fa fa-eye clip flex-none" style={textStyle} />
          <p className="offset-5 flex-none">View Hs Codes</p>
        </div>
      )
    }
    const Viewer = () => {
      if (!viewer) {
        return ''
      }

      return (
        <HsCodeViewer
          item={item}
          hsCodes={hsCodes}
          theme={theme}
          close={this.viewHsCodes}
        />
      )
    }

    const dimensions = dimensionsFn(item)

    return (
      <div className={CONTAINER}>
        <div className="flex-100">
          <h4>Unit {index + 1}</h4>
        </div>

        <hr />

        <div className={LONG_ROW}>
          <p>Gross Weight</p>
          <p>{item.payload_in_kg} kg</p>
        </div>

        <hr className="flex-100" />

        <div className={LONG_ROW}>
          <p>Length</p>
          <p>{item.dimension_y} cm</p>
        </div>

        <hr className="flex-100" />

        <div className={LONG_ROW}>
          <p>Width</p>
          <p>{item.dimension_x} cm</p>
        </div>

        <hr className="flex-100" />

        <div className={LONG_ROW}>
          <p>Height</p>
          <p>{item.dimension_z} cm</p>
        </div>

        <hr className="flex-100" />

        <div className={LONG_ROW}>
          <p>Volume</p>
          <p>
            {dimensions} m<sup>3</sup>
          </p>
        </div>

        <hr className="flex-100" />

        <div className={LONG_ROW}>
          <p>Chargeable Weight</p>
          <p>
            {(item.chargeable_weight).toFixed(2)} kg
          </p>
        </div>

        <hr className="flex-100" />

        {HSCodes()}

        {Viewer()}
      </div>
    )
  }
}
CargoItemDetails.propTypes = {
  item: PropTypes.shape({
    hs_codes: PropTypes.array
  }).isRequired,
  index: PropTypes.number.isRequired,
  viewHSCodes: PropTypes.bool,
  theme: PropTypes.theme,
  hsCodes: PropTypes.arrayOf(PropTypes.string).isRequired
}

CargoItemDetails.defaultProps = {
  viewHSCodes: false,
  theme: false
}

function textStyleFn (theme) {
  return {
    background:
      theme && theme.colors
        ? `-webkit-linear-gradient(left, ${
          theme.colors.primary
        },${
          theme.colors.secondary
        })`
        : 'black'
  }
}

function dimensionsFn (item) {
  const dimensions = item.dimension_y * item.dimension_x * item.dimension_y

  return (dimensions / 1000000).toFixed(2)
}

export default CargoItemDetails
