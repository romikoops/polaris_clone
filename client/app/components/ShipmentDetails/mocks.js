// Will remove it later
// ============================================
import { get } from 'lodash'

export const trucking = { preCarriage: { truckType: '' }, onCarriage: { truckType: '' } }
export const id = 4606
export const selectedDay = '2019-01-24T10:00:00.000Z'

export const ShipmentDetailsAvailableRoutes = [{
  itineraryId: 2863,
  itineraryName: 'Gothenburg - Qingdao',
  modeOfTransport: 'ocean',
  origin: {
    stopId: 4656, hubId: 3023, hubName: 'Gothenburg Port', nexusId: 597, nexusName: 'Gothenburg', latitude: 57.694253, longitude: 11.854048, country: 'SE', truckTypes: ['default']
  },
  destination: {
    stopId: 4657, hubId: 3027, hubName: 'Qingdao Port', nexusId: 601, nexusName: 'Qingdao', latitude: 36.083811, longitude: 120.323534, country: 'CN', truckTypes: []
  }
}, {
  itineraryId: 2958,
  itineraryName: 'Gothenburg - Qingdao',
  modeOfTransport: 'air',
  origin: {
    stopId: 4846, hubId: 3030, hubName: 'Gothenburg Airport', nexusId: 597, nexusName: 'Gothenburg', latitude: 57.694253, longitude: 11.854048, country: 'SE', truckTypes: ['default']
  },
  destination: {
    stopId: 4847, hubId: 3034, hubName: 'Qingdao Airport', nexusId: 601, nexusName: 'Qingdao', latitude: 36.083811, longitude: 120.323534, country: 'CN', truckTypes: []
  }
}]

export const ShipmentDetails = {
  availableRoutes: ShipmentDetailsAvailableRoutes,
  availableMots: ['ocean', 'air']
}

export const cargoItemTypes = [{
  id: 25, dimension_x: null, dimension_y: null, description: 'Pallet', area: null, created_at: '2018-06-27T17:28:28.431Z', updated_at: '2018-06-27T17:28:28.431Z', category: 'Pallet'
}, {
  id: 22, dimension_x: null, dimension_y: null, description: 'Carton', area: null, created_at: '2018-06-27T17:28:28.417Z', updated_at: '2018-06-27T17:28:28.417Z', category: 'Carton'
}, {
  id: 23, dimension_x: null, dimension_y: null, description: 'Crate', area: null, created_at: '2018-06-27T17:28:28.422Z', updated_at: '2018-06-27T17:28:28.422Z', category: 'Crate'
}, {
  id: 26, dimension_x: null, dimension_y: null, description: 'Bottle', area: null, created_at: '2018-06-27T17:28:28.436Z', updated_at: '2018-06-27T17:28:28.436Z', category: 'Bottle'
}, {
  id: 27, dimension_x: null, dimension_y: null, description: 'Stack', area: null, created_at: '2018-06-27T17:28:28.440Z', updated_at: '2018-06-27T17:28:28.440Z', category: 'Stack'
}, {
  id: 28, dimension_x: null, dimension_y: null, description: 'Drum', area: null, created_at: '2018-06-27T17:28:28.445Z', updated_at: '2018-06-27T17:28:28.445Z', category: 'Drum'
}, {
  id: 29, dimension_x: null, dimension_y: null, description: 'Skid', area: null, created_at: '2018-06-27T17:28:28.450Z', updated_at: '2018-06-27T17:28:28.450Z', category: 'Skid'
}, {
  id: 30, dimension_x: null, dimension_y: null, description: 'Barrel', area: null, created_at: '2018-06-27T17:28:28.454Z', updated_at: '2018-06-27T17:28:28.454Z', category: 'Barrel'
}]

export const maxDimensions = {
  general: {
    dimensionX: '590.0', dimensionY: '234.2', dimensionZ: '228.0', payloadInKg: '21770.0', chargeableWeight: '21770.0'
  },
  air: {
    dimensionX: '120.0', dimensionY: '100.0', dimensionZ: '150.0', payloadInKg: '1000.0', chargeableWeight: '1000.0'
  }
}

export const scope = {
  links: { about: '', legal: '' },
  terms: ['You verify that all the information provided above is true', 'You agree to the presented terms and conditions.', 'Demo is to discuss the validity of the presented prices with the product owners.'],
  fee_detail: 'key_and_name',
  closed_shop: false,
  has_customs: true,
  has_insurance: true,
  fixed_currency: false,
  dangerous_goods: false,
  cargo_info_level: 'hs_codes',
  carriage_options: { on_carriage: { export: 'optional', import: 'optional' }, pre_carriage: { export: 'optional', import: 'optional' } },
  detailed_billing: false,
  total_dimensions: true,
  consolidate_cargo: false,
  modes_of_transport: {
    air: { container: true, cargo_item: true }, rail: { container: true, cargo_item: true }, ocean: { container: true, cargo_item: true }, truck: { container: false, cargo_item: false }
  },
  show_beta_features: true,
  closed_registration: false,
  continuous_rounding: false,
  incoterm_info_level: 'text',
  non_stackable_goods: true,
  open_quotation_tool: false,
  customs_export_paper: false,
  fixed_exchange_rates: false,
  require_full_address: true,
  closed_quotation_tool: false
}

export const theme = {
  colors: {
    primary: '#0D5BA9', secondary: '#23802A', brightPrimary: '#2491FD', brightSecondary: '#25ED36'
  },
  logoWide: 'https://assets.itsmycargo.com/assets/logos/Logo_transparent_blue.png',
  logoLarge: 'https://assets.itsmycargo.com/assets/logos/logo_box.png',
  logoSmall: 'https://assets.itsmycargo.com/assets/logos/logo_box.png',
  logoWhite: 'https://assets.itsmycargo.com/assets/logos/Logo_transparent_white.png',
  background: 'https://assets.itsmycargo.com/assets/images/cropped_banner_2.jpg'
}

export const origin = {
  latitude: 57.694253, longitude: 11.854048, nexusId: 597, nexusName: 'Gothenburg', country: 'SE'
}
export const destination = {
  latitude: 36.083811, longitude: 120.323534, nexusId: 601, nexusName: 'Qingdao', country: 'CN'
}
export const cargoItem = {
  payloadInKg: 11, totalVolume: 0, totalWeight: 0, dimensionX: 7, dimensionY: 7, dimensionZ: 8, quantity: 7, cargoItemTypeId: 23, dangerousGoods: false, stackable: false
}

export const cargoItemAggregated = {
  payloadInKg: 0, totalVolume: 122, totalWeight: 346, dimensionX: 0, dimensionY: 0, dimensionZ: 0, quantity: 1, cargoItemTypeId: '', dangerousGoods: false, stackable: true
}

export const cargoItemContainer = {
  sizeClass: 'highCube', quantity: 14, dangerousGoods: false, weight: 16
}

export const cargoUnits = [cargoItem]

export const identity = x => x

export const importedProps = {
  ShipmentDetails,
  aggregatedCargo: false,
  cargoItemTypes,
  cargoUnits,
  destination,
  direction: 'export',
  id,
  loadType: 'cargo_item',
  maxDimensions,
  onCarriage: false,
  onChangeCargoUnitCheckbox: identity,
  onChangeCargoUnitInput: identity,
  onChangeCargoUnitSelect: identity,
  onDeleteUnit: identity,
  onUpdateCargoUnit: identity,
  origin,
  preCarriage: false,
  scope,
  selectedDay,
  theme,
  toggleModal: identity,
  trucking
}

export function logPath (props, path) {
  console.log(path, get(props, path, 'NO SUCH PATH'))
}

export function log (props) {
  let holder = '{\n'
  const declarations = []
  Object.keys(props).forEach((key) => {
    const toLog = `const ${key} = ${JSON.stringify(props[key])}`
    if (toLog.length < 150) console.log(toLog)
    declarations.push(toLog)

    holder = `${holder}\n  ${key},`
  })
  holder = `${holder}\n}`
  console.log(declarations.join('\n'))
  // console.log(holder)
  // console.log(JSON.stringify(props, null, 2))
}
