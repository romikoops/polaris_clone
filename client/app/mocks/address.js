export const firstAddress = {
  user: {
    primary: true
  },
  primary: true,
  id: 246,
  streetNumber: 'FOO_ADDRESS_STREET_NUMBER',
  geocodedAddress: 'FOO_GEOCODED_ADDRESS',
  street: 'FOO_ADDRESS_STREET',
  city: 'FOO_ADDRESS_CITY',
  zipCode: 'FOO_ADDRESS_ZIP_CODE',
  country: 'FOO_ADDRESS_COUNTRY'
}

export const secondAddress = {
  user: {
    primary: false
  },
  primary: false,
  id: 704,
  streetNumber: 'BAR_ADDRESS_STREET_NUMBER',
  geocodedAddress: 'BAR_GEOCODED_ADDRESS',
  street: 'BAR_ADDRESS_STREET',
  city: 'BAR_ADDRESS_CITY',
  zipCode: 'BAR_ADDRESS_ZIP_CODE',
  country: 'BAR_ADDRESS_COUNTRY'
}

export const originAddress = {
  city: 'ORIGIN_ADDRESS_CITY',
  country: 'ORIGIN_ADDRESS_COUNTRY',
  street: 'ORIGIN_ADDRESS_STREET',
  street_number: 'ORIGIN_ADDRESS_STREET_NUMBER',
  zip_code: 'ORIGIN_ADDRESS_ZIP_CODE'
}
export const destinationAddress = {
  city: 'DESTINATION_ADDRESS_CITY',
  country: 'DESTINATION_ADDRESS_COUNTRY',
  street: 'DESTINATION_ADDRESS_STREET',
  street_number: 'DESTINATION_ADDRESS_STREET_NUMBER',
  zip_code: 'DESTINATION_ADDRESS_ZIP_CODE'
}

export const addresses = {
  origin: originAddress,
  destination: destinationAddress
}
