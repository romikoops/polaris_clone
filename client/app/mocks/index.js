export * from './address'
export * from './cards'
export * from './cargoItems'
export * from './client'
export * from './consignee'
export * from './contacts'
export * from './containers'
export * from './currencies'
export * from './dimensions'
export * from './documents'
export * from './feeHash'
export * from './functions'
export * from './gMaps'
export * from './group'
export * from './hub'
export * from './match'
export * from './notifyees'
export * from './offer'
export * from './results'
export * from './routes'
export * from './schedules'
export * from './scope'
export * from './shipmentData'
export * from './shipments'
export * from './shipper'
export * from './tenant'
export * from './theme'
export * from './user'
