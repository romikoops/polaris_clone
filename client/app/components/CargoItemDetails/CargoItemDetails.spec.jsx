import * as React from 'react'
import { mount } from 'enzyme'
import { CargoItemDetails } from './CargoItemDetails'
import { theme } from '../../mocks'

const propsBase = {
  item: {
    payload_in_kg: 56,
    chargeable_weight: 60,
    dimension_x: 111,
    dimension_y: 37,
    dimension_z: 70,
    hs_codes: []
  },
  index: 1,
  viewHSCodes: false,
  theme,
  hsCodes: []
}

let wrapper

const createWrapper = propsInput => mount(<CargoItemDetails {...propsInput} />)

beforeEach(() => {
  wrapper = createWrapper(propsBase)
})

test('unit element use props.index', () => {
  const unitElement = wrapper.find('.flex-100 h4').first()

  expect(unitElement.text().includes(propsBase.index + 1)).toBeTruthy()
})

test('weight element use props.item', () => {
  const weightElement = wrapper.find('p').at(1)

  expect(weightElement.text().includes(propsBase.item.payload_in_kg)).toBeTruthy()
})

test('dimension elements use props.item', () => {
  const dimensionXElement = wrapper.find('p').at(5)
  const dimensionYElement = wrapper.find('p').at(3)
  const dimensionZElement = wrapper.find('p').at(7)

  expect(dimensionXElement.text().includes(propsBase.item.dimension_x)).toBeTruthy()
  expect(dimensionYElement.text().includes(propsBase.item.dimension_y)).toBeTruthy()
  expect(dimensionZElement.text().includes(propsBase.item.dimension_z)).toBeTruthy()
})

test('volume element use props.item', () => {
  const volumeElement = wrapper.find('p').at(9)

  expect(volumeElement.text().includes(0.15)).toBeTruthy()
})

test('chargable weight element use props.item', () => {
  const chargableWeightElement = wrapper.find('p').at(11)

  expect(chargableWeightElement.text().includes(propsBase.item.chargeable_weight)).toBeTruthy()
})

test('initial state doesn\'t include \'View Hs Codes\'', () => {
  expect(wrapper.text().includes('View Hs Codes')).toBeFalsy()
})

test('\'View Hs Codes\' depends on props.viewHSCodes', () => {
  const props = {
    ...propsBase,
    viewHSCodes: true
  }
  const dom = createWrapper(props)

  expect(dom.text().includes('View Hs Codes')).toBeTruthy()
})
