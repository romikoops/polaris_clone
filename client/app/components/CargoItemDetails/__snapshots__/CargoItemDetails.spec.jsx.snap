// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow render 1`] = `
<div
  className="info layout-row flex-100 layout-wrap layout-align-center"
>
  <div
    className="flex-100"
  >
    <h4>
      Unit 
      2
    </h4>
  </div>
  <hr />
  <div
    className="flex-100 layout-row layout-align-space-between"
  >
    <p>
      Gross Weight
    </p>
    <p>
      200
       kg
    </p>
  </div>
  <hr
    className="flex-100"
  />
  <div
    className="flex-100 layout-row layout-align-space-between"
  >
    <p>
      Length
    </p>
    <p>
      60
       cm
    </p>
  </div>
  <hr
    className="flex-100"
  />
  <div
    className="flex-100 layout-row layout-align-space-between"
  >
    <p>
      Width
    </p>
    <p>
      10
       cm
    </p>
  </div>
  <hr
    className="flex-100"
  />
  <div
    className="flex-100 layout-row layout-align-space-between"
  >
    <p>
      Height
    </p>
    <p>
      40
       cm
    </p>
  </div>
  <hr
    className="flex-100"
  />
  <div
    className="flex-100 layout-row layout-align-space-between"
  >
    <p>
      Volume
    </p>
    <p>
      0.04
       m
      <sup>
        3
      </sup>
    </p>
  </div>
  <hr
    className="flex-100"
  />
  <div
    className="flex-100 layout-row layout-align-space-between"
  >
    <p>
      Chargeable Weight
    </p>
    <p>
      250.00
       kg
    </p>
  </div>
  <hr
    className="flex-100"
  />
</div>
`;

exports[`state.viewer is true 1`] = `
<div
  className="info layout-row flex-100 layout-wrap layout-align-center"
>
  <div
    className="flex-100"
  >
    <h4>
      Unit 
      2
    </h4>
  </div>
  <hr />
  <div
    className="flex-100 layout-row layout-align-space-between"
  >
    <p>
      Gross Weight
    </p>
    <p>
      200
       kg
    </p>
  </div>
  <hr
    className="flex-100"
  />
  <div
    className="flex-100 layout-row layout-align-space-between"
  >
    <p>
      Length
    </p>
    <p>
      60
       cm
    </p>
  </div>
  <hr
    className="flex-100"
  />
  <div
    className="flex-100 layout-row layout-align-space-between"
  >
    <p>
      Width
    </p>
    <p>
      10
       cm
    </p>
  </div>
  <hr
    className="flex-100"
  />
  <div
    className="flex-100 layout-row layout-align-space-between"
  >
    <p>
      Height
    </p>
    <p>
      40
       cm
    </p>
  </div>
  <hr
    className="flex-100"
  />
  <div
    className="flex-100 layout-row layout-align-space-between"
  >
    <p>
      Volume
    </p>
    <p>
      0.04
       m
      <sup>
        3
      </sup>
    </p>
  </div>
  <hr
    className="flex-100"
  />
  <div
    className="flex-100 layout-row layout-align-space-between"
  >
    <p>
      Chargeable Weight
    </p>
    <p>
      250.00
       kg
    </p>
  </div>
  <hr
    className="flex-100"
  />
  <HsCodeViewer
    close={[Function]}
    hsCodes={Array []}
    item={
      Object {
        "cargo_item_type_id": "foo",
        "chargeable_weight": 250,
        "dangerous_goods": false,
        "description": "FOO_CARGO_ITEM_DESCRIPTION",
        "dimension_x": 10,
        "dimension_y": 60,
        "dimension_z": 40,
        "gross_weight": 76,
        "hs_codes": Array [
          4,
        ],
        "hs_text": "FOO_CARGO_ITEM_HS_TEXT",
        "id": 1,
        "key": "FOO_CARGO_ITEM_KEY",
        "payload_in_kg": 200,
        "quantity": 5,
        "stackable": true,
        "tare_weight": 20,
        "weight": 220,
      }
    }
    t={[Function]}
    theme={
      Object {
        "colors": Object {
          "primary": "#333",
          "secondary": "#fafafa",
        },
      }
    }
  />
</div>
`;

exports[`theme.colors is falsy 1`] = `
<div
  className="info layout-row flex-100 layout-wrap layout-align-center"
>
  <div
    className="flex-100"
  >
    <h4>
      Unit 
      2
    </h4>
  </div>
  <hr />
  <div
    className="flex-100 layout-row layout-align-space-between"
  >
    <p>
      Gross Weight
    </p>
    <p>
      200
       kg
    </p>
  </div>
  <hr
    className="flex-100"
  />
  <div
    className="flex-100 layout-row layout-align-space-between"
  >
    <p>
      Length
    </p>
    <p>
      60
       cm
    </p>
  </div>
  <hr
    className="flex-100"
  />
  <div
    className="flex-100 layout-row layout-align-space-between"
  >
    <p>
      Width
    </p>
    <p>
      10
       cm
    </p>
  </div>
  <hr
    className="flex-100"
  />
  <div
    className="flex-100 layout-row layout-align-space-between"
  >
    <p>
      Height
    </p>
    <p>
      40
       cm
    </p>
  </div>
  <hr
    className="flex-100"
  />
  <div
    className="flex-100 layout-row layout-align-space-between"
  >
    <p>
      Volume
    </p>
    <p>
      0.04
       m
      <sup>
        3
      </sup>
    </p>
  </div>
  <hr
    className="flex-100"
  />
  <div
    className="flex-100 layout-row layout-align-space-between"
  >
    <p>
      Chargeable Weight
    </p>
    <p>
      250.00
       kg
    </p>
  </div>
  <hr
    className="flex-100"
  />
</div>
`;

exports[`viewHSCodes is true 1`] = `
<div
  className="info layout-row flex-100 layout-wrap layout-align-center"
>
  <div
    className="flex-100"
  >
    <h4>
      Unit 
      2
    </h4>
  </div>
  <hr />
  <div
    className="flex-100 layout-row layout-align-space-between"
  >
    <p>
      Gross Weight
    </p>
    <p>
      200
       kg
    </p>
  </div>
  <hr
    className="flex-100"
  />
  <div
    className="flex-100 layout-row layout-align-space-between"
  >
    <p>
      Length
    </p>
    <p>
      60
       cm
    </p>
  </div>
  <hr
    className="flex-100"
  />
  <div
    className="flex-100 layout-row layout-align-space-between"
  >
    <p>
      Width
    </p>
    <p>
      10
       cm
    </p>
  </div>
  <hr
    className="flex-100"
  />
  <div
    className="flex-100 layout-row layout-align-space-between"
  >
    <p>
      Height
    </p>
    <p>
      40
       cm
    </p>
  </div>
  <hr
    className="flex-100"
  />
  <div
    className="flex-100 layout-row layout-align-space-between"
  >
    <p>
      Volume
    </p>
    <p>
      0.04
       m
      <sup>
        3
      </sup>
    </p>
  </div>
  <hr
    className="flex-100"
  />
  <div
    className="flex-100 layout-row layout-align-space-between"
  >
    <p>
      Chargeable Weight
    </p>
    <p>
      250.00
       kg
    </p>
  </div>
  <hr
    className="flex-100"
  />
  <div
    className="flex-100 layout-row layout-wrap"
    onClick={[Function]}
  >
    <i
      className="fa fa-eye clip flex-none"
      style={
        Object {
          "background": "-webkit-linear-gradient(left, #333,#fafafa)",
        }
      }
    />
    <p
      className="offset-5 flex-none"
    >
      View Hs Codes
    </p>
  </div>
</div>
`;
